REGRESSION(253824@main): [GTK][WPE] ASSERTION FAILED: m_sessionID.isEphemeral() || !path.isEmpty() running C API tests

carlosgcampos opened the pull request:
#### 102fc4e9903836a4beac1b549c8dc42419ae422b
<pre>
REGRESSION(253824@main): [GTK][WPE] ASSERTION FAILED: m_sessionID.isEphemeral() || !path.isEmpty() running C API tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=252473">https://bugs.webkit.org/show_bug.cgi?id=252473</a>

Reviewed by Michael Catanzaro.

The problem is that the default website data store is created with an
ephemeral configuration, but a persistent session ID.

* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::defaultDataStoreIsPersistent):
(WebKit::WebsiteDataStore::defaultDataStore):

Canonical link: <a href="https://commits.webkit.org/260445@main">https://commits.webkit.org/260445@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/043c811b2f9c1d439baa9c053758c4dc7d017ff7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108329 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17421 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41193 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117445 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116823 "Built successfully") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/18935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8700 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100547 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114098 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/18935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97367 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42095 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96107 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29011 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83775 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10257 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30355 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10997 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7264 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16407 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49952 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7226 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12583 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
That's a long time to not notice failing tests. O_O

carlosgcampos committed:
REGRESSION(253824@main): [GTK][WPE] ASSERTION FAILED: m_sessionID.isEphemeral() || !path.isEmpty() running C API tests
https://bugs.webkit.org/show_bug.cgi?id=252473

Reviewed by Michael Catanzaro.

The problem is that the default website data store is created with an
ephemeral configuration, but a persistent session ID.

* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::defaultDataStoreIsPersistent):
(WebKit::WebsiteDataStore::defaultDataStore):

Canonical link: https://commits.webkit.org/260445@main
